packages/admin/admin/src/submissionBoundary/SubmissionBoundary.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from "react";
+import { useIntl } from "react-intl";
+import useConstant from "use-constant";
+import { v4 as uuid } from "uuid";
+
+import { messages } from "../messages";
+import { RouterPrompt } from "../router/Prompt";
+
+export type SaveActionSuccess = boolean;
+export interface SubmissionBoundaryApi {
+    save: () => Promise<SaveActionSuccess>;
+    register: (id: string, props: SubmissionBoundaryStateProps) => void;
+    unregister: (id: string) => void;
```

Comment:
Should we introduce the concept of "submittable" for parts of the submission boundary that can be submitted? Like `registerSubmittable()`, `unregisterSubmittable()` and `<Submittable hasChanges doSave />`?

This would be easier if we'd use SaveBoundary and Savable 🤔

Reply:
register/unregister is in the context of SubmissionBoundaryApi, imho that naming is ok.

`<Submittable` is a good idea, this is really not a state in the react meaning of state

> This would be easier if we'd use SaveBoundary and Savable  

I don't care :)

Reply:
> register/unregister is in the context of SubmissionBoundaryApi, imho that naming is ok.

Fine by me, let's keep the naming 

> > This would be easier if we'd use SaveBoundary and Savable
> 
> I don't care :)

I'd say we rename to SaveBoundary and Savable. We're already using save for the action, the button, the saving state. Seems more in line for me

Reply:
Renamed: SubmissionBoundary -> SaveBoundary, SubmissionBoundaryState -> Saveable